packages/validator-ajv8/src/types.ts

```diff
@@ -21,3 +22,17 @@ export interface CustomValidatorOptionsType {
 /** The type describing a function that takes a list of Ajv `ErrorObject`s and localizes them
  */
 export type Localizer = (errors?: null | ErrorObject[]) => void;
+
+/** Extend ValidateFunction to Omit its two required properties, `schema` and `schemaEnv` that are not produced by the
+ * AJV schema standalone compilation code
+ */
+export interface CompiledValidateFunction<T = any> extends Omit<ValidateFunction<T>, 'schema' | 'schemaEnv'> {
+  /** This is literally copied from the `ValidateFunction` type definition from which it extends because it seems to get
+   * lost as part of the Omit<>.
+   */
+  (this: Ajv | any, data: any, dataCxt?: DataValidationCxt): boolean;
+}
```

Comment:
Tbh I don't really know how a function definition is supposed to work with `Omit`, but it makes sense to do this, since it also has properties you want to drop.